src/Confluent.Kafka/Timestamp.cs

```diff
@@ -36,16 +36,38 @@ public struct Timestamp
         /// <summary>
         ///     Initializes a new instance of the Timestamp structure.
         /// </summary>
+        /// <param name="unixTimestampMs">
+        ///     The unix millisecond timestamp.
+        /// </param>
+        /// <param name="type">
+        ///     The type of the timestamp.
+        /// </param>
+        public Timestamp(long unixTimestampMs, TimestampType type)
+        {
+            Type = type;
+            UnixTimestampMs = unixTimestampMs;
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the Timestamp structure.
+        ///     Prefer the (long, TimestampType) overload if possible.
```

Comment:
Ok - i see where you're coming from. I think having the `DateTime` constructor will be convenient when people are specifying timestamps to `ProduceAsync`. I'd be happy to depreciate the `DateTime` property - I don't have a strong opinion on that, but maybe removing it is slightly better.